server/controllers/finance/reports/invoicedReceivedStock/index.js

```diff
@@ -0,0 +1,181 @@
+const _ = require('lodash');
+const db = require('../../../../lib/db');
+const ReportManager = require('../../../../lib/ReportManager');
+const Patients = require('../../../medical/patients');
+
+const TEMPLATE = './server/controllers/finance/reports/invoicedReceivedStock/report.handlebars';
+// expose to the API
+exports.report = report;
+
+// default report parameters
+const DEFAULT_PARAMS = {
+  csvKey : 'invoiced_received_stock',
+  filename : 'REPORT.COMPARE_INVOICED_RECEIVED.TITLE',
+  orientation : 'landscape',
+};
+
+/**
+ * @function report
+ *
+ * @description
+ * This report makes it possible to attach the items billed to a patient on
+ * the articulated items consumed or distributed to the Patient.
+ */
+function report(req, res, next) {
+  const params = req.query;
+  const patientUuid = req.params.uuid;
+
+  const data = {};
+  let reporting;
+
+  params.dateFrom = new Date(params.dateFrom);
+  params.dateTo = new Date(params.dateTo);
+
+  data.period = {
+    dateFrom : params.dateFrom,
+    dateTo : params.dateTo,
+  };
+
+  const sqlInvoice = `
+    SELECT iv.uuid, map.text, iv.date
+    FROM invoice AS iv
+    JOIN debtor AS d ON d.uuid = iv.debtor_uuid
+    JOIN patient AS p ON p.debtor_uuid = d.uuid
+    JOIN document_map AS map ON map.uuid = iv.uuid
+    WHERE p.uuid = ? AND (DATE(iv.date) >= DATE(?) AND DATE(iv.date) <= DATE(?))
+    ORDER BY iv.date DESC
+  `;
+
+  const sqlInvoicedDistributed = `
+  SELECT aggr.reference, BUID(aggr.inventory_uuid) AS inventory_uuid, aggr.inventory_text,
+  SUM(aggr.quantity) AS quantity_invoiced, aggr.price_invoiced,
+  BUID(aggr.invoice_uuid) AS invoice_uuid, aggr.invoice_date,
+  SUM(aggr.quantity_distributed) AS quantity_distributed, SUM(aggr.cost_distributed) AS cost_distributed
+  FROM (
+    SELECT map.text AS reference, inv.uuid AS inventory_uuid, inv.text AS inventory_text, item.quantity,
+    item.inventory_price AS price_invoiced,iv.uuid AS invoice_uuid, iv.date AS invoice_date, 0 AS quantity_distributed,
+    0 AS cost_distributed
+    FROM invoice_item AS item
+    JOIN inventory AS inv ON inv.uuid = item.inventory_uuid
+    JOIN invoice AS iv ON iv.uuid = item.invoice_uuid
+    JOIN document_map AS map ON map.uuid = iv.uuid
+    JOIN patient AS p ON p.debtor_uuid = iv.debtor_uuid
+    WHERE p.uuid = ? AND (DATE(iv.date) >= DATE(?) AND DATE(iv.date) <= DATE(?))
+    AND inv.consumable = 1
+    UNION
+    SELECT map.text AS reference, inv.uuid AS inventory_uuid, inv.text AS inventory_text,
+    0 AS quantity, 0 AS price_invoiced, sm.invoice_uuid, sm.date,
+    SUM(sm.quantity) AS quantity_distributed, sm.unit_cost AS cost_distributed
+    FROM stock_movement AS sm
+    JOIN patient AS p ON p.uuid = sm.entity_uuid
+    JOIN lot AS l ON l.uuid = sm.lot_uuid
+    JOIN inventory AS inv ON inv.uuid = l.inventory_uuid
+    JOIN invoice AS iv ON iv.uuid = sm.invoice_uuid
+    JOIN document_map AS map ON map.uuid = iv.uuid
+    WHERE sm.invoice_uuid IS NOT NULL AND
```

Comment:
Do you need this `sm.invoice_uuid IS NOT NULL`?  I think the `JOIN` on inventory should already take care of that, no?